streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -163,8 +164,8 @@ public void register(StateStore store, RestoreFunc restoreFunc) {
         long newOffset = restoreConsumer.position(storePartition);
         restoredOffsets.put(storePartition, newOffset);
 
-        // un-subscribe the change log partition
-        restoreConsumer.subscribe(Collections.<String>emptyList());
+        // un-assign the change log partition
+        restoreConsumer.assign(Collections.<TopicPartition>emptyList());
```

Comment:
There is a new API added along with regex subscription patch:

unsubscribe()

which unsubscribed all current subscriptions, could you merge from apache trunk and use this function instead?